tools/xabuild/xabuild.csproj

```diff
@@ -37,6 +37,9 @@
     <Reference Include="MSBuild">
       <HintPath>$(MSBuildReferencePath)\MSBuild.$(_MSBuildExtension)</HintPath>
     </Reference>
+    <Reference Include="System.Reflection.Metadata" Condition=" '$(OS)' != 'Windows_NT' ">
```

Comment:
What happens on Windows?  Why isn't this needed there?

> it breaks the build on macOS to remove [the `@(Reference)` to `System.Reflection.Metadata`].

What's the exact breakage?

`monodis --assemblyref bin/Debug/bin/xabuild.exe` doesn't show any references from `xabuild.exe` to `System.Reflection.Metadata.dll`, so I do not understand why this `@(Reference)` would be needed at all.

That said, I *do*  have a `bin/Debug/bin/System.Reflection.Metadata.dll`, so *something* copied it into `bin/Debug/bin` -- presumably this `@(Reference)` from earlier (I haven't cleaned `bin/Debug/bin` recently) -- but it's entirely possible that having `bin/Debug/bin/System.Reflection.Metadata.dll` is the actual problem...

Reply:
It appears that `C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\System.Reflection.Metadata.dll` does not exist on Windows, and none of the MSBuild assemblies reference it.

Reply:
On macOS `MSBuild.dll` references `System.Reflection.Metadata.dll`.

Reply:
After some discussion, trying the `AppDomain.CurrentDomain.AssemblyResolve` event.